layout/cycle_remover.go

```diff
@@ -16,34 +16,32 @@ func NewSimpleCycleRemover() SimpleCycleRemover {
 	}
 }
 
-func getCycleDFS(g Graph, que []uint64) []uint64 {
+func getCycleDFS(neighbors map[uint64][]uint64, que []uint64) []uint64 {
```

Comment:
good to have slow and versions. old one looks correct. then they can validate each other in tests
```suggestion
func getCycleDFSViaNeighbours(neighbors map[uint64][]uint64, que []uint64) []uint64 {
```